src/features/bivariate_manager/components/BivariateMatrixContainer/BivariateMatrixContainer.tsx

```diff
@@ -65,18 +69,31 @@ const BivariateMatrixContainer = ({ className }: BivariateMatrixContainerProps)
           ),
           ready: () => (
             <>
+              <PanelHeader />
               <div className={s.matrixContainer}>
                 <ConnectedBivariateMatrix ref={onRefChange} />
               </div>
               <BivariateGreetingsContainer className={s.greetings} />
             </>
           ),
         })}
-        <div className={s.topRightCorner} />
-        <div className={s.bottomRightCorner} />
       </div>
     </div>
   );
 };
 
+const PanelHeader = () => {
+  const [focusedGeometry] = useAtom(focusedGeometryAtom);
+  const haveGeometry = !isGeometryEmpty(focusedGeometry);
+
+  return (
+    <div className={s.header}>
+      <Text type="heading-l">{i18n.t('bivariate.matrix.header.title')}</Text>
+      {haveGeometry && (
+        <div className={s.hint}>{i18n.t('bivariate.matrix.header.hint')}</div>
+      )}
+    </div>
+  );
+};
+
```

Comment:
why do you declare separate function component for that?
even if you do, it would be good to see it in a separate file

Reply:
to keep main component from growing
if not to do this it will become a big hardly readable mess in few iterations

I felt like it's too small to separate

Reply:
Hm may be it is better to read in a separate component